source/puppet/4.2/reference/release_notes_agent.md

```diff
@@ -0,0 +1,127 @@
+---
+layout: default
+title: "Puppet Agent 1.2.x Release Notes"
+canonical: "/puppet/latest/reference/release_notes_agent.html"
+---
+
+[Puppet 4.2.0]: /puppet/4.2/reference/release_notes.html#puppet-420
+[Puppet 4.2.1]: /puppet/4.2/reference/release_notes.html#puppet-421
+[Puppet 4.2.2]: /puppet/4.2/reference/release_notes.html#puppet-422
+[Puppet 4.2.3]: /puppet/4.2/reference/release_notes.html#puppet-423
+[Facter 3.0.0]: /facter/3.0/reference/release_notes.html#facter-300
+[Facter 3.0.1]: /facter/3.0/reference/release_notes.html#facter-301
+[Facter 3.0.2]: /facter/3.0/reference/release_notes.html#facter-302
+[Facter 3.1.0]: /facter/3.1/reference/release_notes.html#facter-310
+[Facter 3.1.1]: /facter/3.1/reference/release_notes.html#facter-311
+[Hiera 3.0.1]: /hiera/3.0/release_notes.html#hiera-301
+[Hiera 3.0.3]: /hiera/3.0/release_notes.html#hiera-303
+[Hiera 3.0.4]: /hiera/3.0/release_notes.html#hiera-304
+[MCollective 2.8.6]: /mcollective/releasenotes.html#2_8_6
+[MCollective 2.8.5]: /mcollective/releasenotes.html#2_8_5
+[MCollective 2.8.2]: /mcollective/releasenotes.html#2_8_2
+
+This page lists changes to the `puppet-agent` package. For details about changes to components in a `puppet-agent` release, follow the links to those components in the package release's notes.
+
+The `puppet-agent` package's version numbers use the format X.Y.Z, where:
+
+* X must increase for major backwards-incompatible changes
+* Y may increase for backwards-compatible new functionality
+* Z may increase for bug fixes
+
+## If You're Upgrading from Puppet 3.x
+
+The `puppet-agent` package installs Puppet 4. Also read the [Puppet 4.0 release notes](/puppet/4.0/reference/release_notes.html), since they cover any breaking changes since Puppet 3.8.
+
+Also of interest: [About Agent](/puppet/4.2/reference/about_agent.html) and the [Puppet 4.2 release notes](/puppet/4.2/reference/release_notes.html).
+
+## 1.2.7
```

Comment:
Okay, and I know this is the dumbest thing ever, but: Please use "### Puppet Agent 1.2.7" for these headers instead of the version number alone. 

Kramdown, our MD renderer, will not produce usable automatic header IDs from headers that don't begin with at least one letter; it'll just produce unpredictable garbage. To ensure we can reliably link to each version, we need to put words in front of the numbers. 

I know, I'm sorry. >:|